auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -231,48 +224,86 @@ public static int getDefaultTheme() {
      * fails if the Firebase Auth deletion fails. Credentials deletion failures are handled
      * silently.
      *
-     * @param activity the calling {@link Activity}.
+     * @param context the calling {@link Context}.
      */
     @NonNull
-    public Task<Void> delete(@NonNull FragmentActivity activity) {
-        // Initialize SmartLock helper
-        GoogleSignInHelper signInHelper = GoogleSignInHelper.getInstance(activity);
-
-        FirebaseUser firebaseUser = FirebaseAuth.getInstance().getCurrentUser();
-        if (firebaseUser == null) {
+    public Task<Void> delete(@NonNull Context context) {
+        FirebaseUser currentUser = FirebaseAuth.getInstance().getCurrentUser();
+        if (currentUser == null) {
             // If the current user is null, return a failed task immediately
-            return Tasks.forException(new Exception("No currently signed in user."));
+            return Tasks.forException(new FirebaseAuthInvalidUserException(
+                    String.valueOf(CommonStatusCodes.SIGN_IN_REQUIRED),
+                    "No currently signed in user."));
         }
 
         // Delete the Firebase user
-        Task<Void> deleteUserTask = firebaseUser.delete();
+        Task<Void> deleteUserTask = currentUser.delete();
 
         // Get all SmartLock credentials associated with the user
-        List<Credential> credentials = SmartLockBase.credentialsFromFirebaseUser(firebaseUser);
+        List<Credential> credentials = credentialsFromFirebaseUser(currentUser);
 
+        CredentialsClient client;
+        if (context instanceof Activity) {
+            client = Credentials.getClient((Activity) context);
+        } else {
+            client = Credentials.getClient(context);
+        }
         // For each Credential in the list, create a task to delete it.
         List<Task<?>> credentialTasks = new ArrayList<>();
         for (Credential credential : credentials) {
-            credentialTasks.add(signInHelper.delete(credential));
+            credentialTasks.add(client.delete(credential));
         }
 
         // Create a combined task that will succeed when all credential delete operations
         // have completed (even if they fail).
         final Task<Void> combinedCredentialTask = Tasks.whenAll(credentialTasks);
 
-        // Chain the Firebase Auth delete task with the combined Credentials task
-        // and return.
-        return deleteUserTask.continueWithTask(new Continuation<Void, Task<Void>>() {
-            @Override
-            public Task<Void> then(@NonNull Task<Void> task) throws Exception {
-                // Call getResult() to propagate failure by throwing an exception
-                // if there was one.
-                task.getResult(Exception.class);
+        return Tasks.whenAll(deleteUserTask, signOutIdps(context), combinedCredentialTask);
```

Comment:
This is a slight behavior change.  Before we tried to propagate user deletion failure exceptions explicitly by checking the result of that task first.  We used `continueWithTask` not `whenAll`.

Looking at this now, it may also be the case that it would be better to wait for the user `delete` to succeed before doing the credential deletion, but I don't feel strongly about that since I have never seen a `delete` fail.

Reply:
Ohhhhhh, I see what was going on there: it prevents that "1 out of n underlying tasks failed" exception. Devs can still get the underlying exception with a `getCause`, but would you like to restore previous behavior?